src/plugins/intel_gpu/src/kernel_selector/cl_kernels/gemm_tiled_opt.cl

```diff
@@ -86,18 +87,26 @@ KERNEL(gemm_tiled_opt)(
 
     // Setting x and y for fusings indexing
     // TODO: investigate how we can use only TILE_N_NOT_DIVISIBLE here for getting stable results in fusings
+#if IS_DYNAMIC
+    const uint x = (uint)get_global_id(0);
+#else // IS_DYNAMIC
 #if TILE_N_NOT_DIVISIBLE || B_VEC_SIZE == 1
     const uint x = (uint)get_global_id(0);
 #else // TILE_N_NOT_DIVISIBLE || B_VEC_SIZE == 1
     const uint x = tile_n_num * SIMD_WIDTH * B_VEC_SIZE;
 #endif // TILE_N_NOT_DIVISIBLE || B_VEC_SIZE == 1
+#endif // IS_DYNAMIC
     uint y = tile_m_offset;
 
+#if IS_DYNAMIC
+    const uint tile_m_iterations = TILE_M_NOT_DIVISIBLE ? (tile_m_num == (tile_m_size - 1) ? TILE_M_LEFTOVER : TILE_M) : TILE_M;
```

Comment:
Can we have this version for all cases (including static ones)? I'd expect that the compiler will emit the same code for
```
const uint tile_m_iterations = TILE_M_NOT_DIVISIBLE ? (tile_m_num == (tile_m_size - 1) ? TILE_M_LEFTOVER : TILE_M) : TILE_M;
```

and 

```
#if TILE_M_NOT_DIVISIBLE
    const uint tile_m_iterations = tile_m_num == (tile_m_size - 1) ? TILE_M_LEFTOVER : TILE_M;
#else // TILE_M_NOT_DIVISIBLE
    const uint tile_m_iterations = TILE_M;
#endif // TILE_M_NOT_DIVISIBLE
```

when `TILE_M_NOT_DIVISIBLE` is compile-time constant

Reply:
For the same reason as above, this code was separated.

Reply:
I understand your suggestion. removed unnecessarily redundant code as suggested.

Reply:
Thanks. But please check the performance in static case to ensure that compiler emits the same code.